contents/java/streams-an-introduction.md

```diff
@@ -0,0 +1,75 @@
+
+It's okay if you have no experience in writing lambda expressions! This guide will step you through the basics of both lambda expressions and streams so that you can start utilising them.
+
+## Getting Started
+
+Before we can get started on Stream, it is imperative to first have some understanding on the concept of functional interface and lambda expressions. Feel free to skip right ahead if you are already familiar with such concepts!
+
+### Functional Interface and Lambda Expressions
+
+If you have tried using the Stream library and looked at its [API page](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html), you would see that many of its methods accept objects of new types such as `Predicate`, `Collector` and more. If you navigate into the APIs of these new types, you would have realised they are merely interfaces (and the APIs themselves aren't giving much help on how you are supposed to make use of these interfaces).
+
+These interfaces are actually functional interfaces, which are simply __interfaces declared with a single abstract method__ (, excluding abstract methods which overrides public methods of `java.lang.Object`). These interfaces provide an easy way for you to somehow reference a method, which could be a existing class method, class constructor or a lambda expression.
+
+Let's see this through an example.
+
+`Comparator<T>` may be a familiar interface to you. In previous Java editions, to create an object extending this interface, you could either instantiate the object from a named class or an anonymous class, both of which need to override the abstract methods in `Comparator<T>`. So the code may look something like this:
+```
+// trying to sort students by height
+students.sort(new Comparator<Student>() {
+    public int compare(Student s1, Student s2) {
+      return Double.compare(s1.getHeight(), s2.getHeight());
+    }
+  });
+```
+Let's face it. Declaring classes (anonymous or not) is quite a pain. Declaring a named class adds to the number of classes you need to maintain while the syntax of a anonymous class is quite an eyesore.
+
+In Java 8, this interface has become a functional interface (surprise, surprise), which allows you to write the code like this instead!
+```
+// trying to sort students by height
+students.sort((s1, s2) -> Double.compare(s1.getHeight(), s2.getHeight()));
+```
+The compiler is able to infer that a object of type `Comparator<Student>` is expected and that the lambda expression fits into the definition for `compare` (the single abstract method), thus creating an instance of type `Comparator<Student>` with the lambda expression implemented as the `compare` method.
+
+### The Stream API
+
+## Using Java 8 Streams
+
+### Drawbacks and Pitfalls
+
+## Resources
```

Comment:
You can consider linking any reference materials used to write the book chapter, or additional materials that introduce other features of Stream.